src/osd/PrimaryLogPG.cc

```diff
@@ -1713,6 +1713,17 @@ void PrimaryLogPG::do_op(OpRequestRef& op)
   hobject_t head = m->get_hobj();
   head.snap = CEPH_NOSNAP;
 
+  if (!info.pgid.pgid.contains(
+	info.pgid.pgid.get_split_bits(pool.info.get_pg_num()), head)) {
```

Comment:
Hi @liewegas , I am using kernel rbd (kernel 3.10 in RHEL 7.4) as a client for a 12.2.12 ceph cluster. But I found an error message as below:
```
2019-07-02 03:58:43.724305 osd.3 osd.3 192.168.15.3:6800/37773 1 : cluster [WRN] 6.0 does not contain 6:06a674d9:::rbd_data.11d36b8b4567.0000000000000023:head op osd_op(client.4615.1:286132 6.9b2e6560 6:06a674d9:::rbd_data.11d36b8b4567.0000000000000023:head [set-alloc-hint object_size 4194304 write_size 4194304,write 0~4096] snapc 0=[] ondisk+write+known_if_redirected e95) v4
```
meanwhile, a request in client is hanging and never to be complete.:
```
REQUESTS 1 homeless 0
286132    osd3    6.9b2e6560    [3,5,1]/3    [3,5,1]/3    rbd_data.11d36b8b4567.0000000000000023    0x400024    1    0'0    set-alloc-hint,write
```

At that time, we are doing pg splitting, and yes, the client op epoch is  e95 but server epoch is already e99. 

But we reach this warning because the kernel client in RHEL 7.4 is not supporting CEPH_FEATURE_OSD_POOLRESEND. So it was not dropped in can_discard_op():
```
  } else if (m->get_connection()->has_feature(CEPH_FEATURE_OSD_POOLRESEND)) {
    // < luminous client
    if (m->get_map_epoch() < pool.info.get_last_force_op_resend_preluminous()) {
      dout(7) << __func__ << " sent before last_force_op_resend_preluminous "
	      << pool.info.last_force_op_resend_preluminous
	      << ", dropping" << *m << dendl;
      return true;
    }
  }
```
My question is, what should we do in this case, when client is not supporting CEPH_FEATURE_OSD_POOLRESEND? 
 assert on that to force the client resend?
```
if (!m->get_connection()->has_feature(CEPH_FEATURE_OSD_POOLRESEND))
    assert(1)
```